src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1195,6 +1223,100 @@ def construct_payload(self):
         super().construct_payload()
         self.set_up_service_binds()
         self.set_up_extended_location()
+        self.set_up_source()
+        self.set_up_repo()
+
+    def set_up_source(self):
+        if self.get_argument_source():
+            self._set_up_source_or_repo()
+
+    def set_up_repo(self):
+        if self.get_argument_repo():
+            container_app = self._get_containerapp_if_exists()
+            if container_app:
+                # Executing the create command with --repo makes two ARM calls. The container app is created first
+                # and the source control is set only after the container app is created. This means, when the container app is
+                # first created, the image would be defaulted to mcr.microsoft.com/k8se/quickstart:latest (if --image was not provided)
+                # and the image name would change to `registry-server/containerapp-name:{github-sha}` only
+                # after the source control completes successfully
+                # and the new image is deployed.
+                # If the create command is executed multiple times however, we need to make sure that the image is not defaulted
+                # to mcr.microsoft.com/k8se/quickstart:latest if source control fails the subsequent times by setting the image
+                # to the previously deployed image.
+
+                if container_app["properties"]["template"]["containers"] is None or len(container_app["properties"]["template"]["containers"]) == 0:
+                    raise ValidationError(
```

Comment:
If the `container_app` does not have any containers, if the cx re-try will `--repo` and `--image`, it will continue run into this error, right?

Reply:
Firstly, don't think it is even possible for this scenario to occur, just because a container app cannot be created without a container. 
It the container app already exists, it should have an image associated with it and don't think cx can remove any containers? they can only update or add more side containers.
But if this error occurred hypothetically, the cx would have to re-try with --image first to ensure there is an image associated with the container app and re-run `az containerapp create` with `--repo`